### PR TITLE
Amend/fix get todo by id route

### PR DESCRIPTION
**Pull request by lanastasi-hiyield**
This PR fixes the following:

- Getting individual TODO item not working, replaced .some() with filter()

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -33,9 +33,10 @@ app.get('/todo-item', function(request, response) {
 
 // Get todo-item by id
 app.get('/todo-item/:id', function(request, response) {
-  let requestedId = todos.some(e => e.id == request.params.id);
+  let requestedId = todos.filter(todo => todo.id == request.params.id)[0];
+  console.log(requestedId);
```
Comment (lfiore18) [inline app.js]: REMOVE THIS CONSOLE LOG!

File: app.js [hunk 2]
```diff
@@ -33,9 +33,10 @@ app.get('/todo-item', function(request, response) {
...
   if (requestedId) {
-    response.status(200).json(todos[request.params.id]);
+    response.status(200).json(requestedId);
   }
```
Comment (lfiore18) [inline app.js:42]: Uh, no status response for a failure?

File: main.js [hunk 3]
```diff
@@ -43,10 +43,9 @@ async function getTodoById(id) {
 }
 
 async function apiCall(endPoint) {
-  return await fetch(`${url+endPoint}`
-  )
-    .then(response => response.json())
-    .then(json => json);
+  let response =  await fetch(`${url+endPoint}`);
+  response = await response.json();
+  return response;
```
Comment (lfiore18) [inline main.js:48]: Nice use of await and async!

**Review by lfiore18 — CHANGES_REQUESTED** [review]
Change the things I asked you for!

**Review by AnitaMcMillan — APPROVED** [review]
Lovely work!

**Review by lfiore18 — APPROVED** [review]
Hmmph, alright I guess.

**Pull request merged**